editor/src/components/editor/actions/actions.tsx

```diff
@@ -4771,15 +4775,38 @@ export const UPDATE_FNS = {
           }
         },
       )
-      const updatedEditorState: EditorModel = {
-        ...withNewElement,
-        selectedViews: newSelectedViews,
-        trueUpGroupsForElementAfterDomWalkerRuns: [
-          ...editor.trueUpGroupsForElementAfterDomWalkerRuns,
-          EP.appendToPath(action.insertionPath.intendedParentPath, newUID),
-        ],
+
+      let groupCommands: CanvasCommand[] = []
+      if (treatElementAsGroupLike(editor.jsxMetadata, action.insertionPath.intendedParentPath)) {
+        const group = MetadataUtils.findElementByElementPath(
+          editor.jsxMetadata,
+          action.insertionPath.intendedParentPath,
+        )
+        if (group != null && action.toInsert.element.type === 'JSX_ELEMENT') {
+          // this condition would need updating when we can insert fragments or conditionals with children
```

Comment:
I would suggest making this throw an exception for things that aren't `JSX_ELEMENT`, otherwise we'll just pass right over this when we do support that.

Reply:
https://github.com/concrete-utopia/utopia/pull/4004/commits/d578af2083636126e1df32359fecc18b413f1c09